src/Polysemy/ConstraintAbsorber/MonadError.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE FlexibleInstances           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE MultiParamTypeClasses       #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE UndecidableInstances        #-}
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module Polysemy.ConstraintAbsorber.MonadError
+  (
+    absorbError
+  )
+where
+
+import           Polysemy
+import           Polysemy.ConstraintAbsorber
+import           Polysemy.Error
+import qualified Control.Monad.Error.Class as S
+
+------------------------------------------------------------------------------
+-- | absorb a @MonadError e@ constraint into @Member (Error e) r => Sem r@
+absorbError :: Member (Error e) r
+  => (S.MonadError e (Sem r) => Sem r a) -> Sem r a
+absorbError = absorbWithSem @(S.MonadError _) @Action
+  (ErrorDict throw catch)
+  (Sub Dict) 
+{-# INLINEABLE absorbError #-}
+
+-- | A dictionary of the functions we need to supply
+-- to make an instance of Error
+data ErrorDict e m = ErrorDict
+  {
+    throwError_ :: forall a. e -> m a
+  , catchError_ :: forall a. m a -> (e -> m a) -> m a
+  }
+
+-- | Wrapper for a monadic action with phantom
+-- type parameter for reflection.
+-- Locally defined so that the instance we are going
+-- to build with reflection must be coherent, that is
+-- there cannot be orphans.
+newtype Action m s' a = Action { action :: m a } deriving (Functor, Applicative, Monad)
```

Comment:
It might be clearer to name this `Orphan`.

Reply:
It might be.  But I hate it.  I could be convinced.  And it doesn't really matter since we aren't exporting it.  ```Orphan``` speaks to part of it's function but not all of it.  And ```Action``` speaks to a different part.  Both seem better than ```Lift``` to me.  ```OrphanAction```?

Reply:
I'm not going to fight you on this! I just name newtypes `Orphan` whenever they're necessary to avoid orphan instances and I'm not going to export them anyway!

Reply:
Let's leave it then.  Especially since we can always change it without affecting anything else at all.